crates/rolldown_plugin/src/plugin_context.rs

```diff
@@ -67,6 +101,27 @@ impl From<PluginContextImpl> for PluginContext {
 }
 
 impl PluginContextImpl {
+  pub async fn load(
+    &self,
+    specifier: &str,
+    load_callback_fn: Box<LoadCallbackFn>,
+  ) -> anyhow::Result<()> {
+    self.context_load_modules.insert(specifier.into(), LoadCallback(Box::new(load_callback_fn)));
+    self
+      .tx
+      .send(ModuleLoaderMsg::FetchModule(ResolvedId {
+        id: specifier.into(),
+        ignored: false,
+        module_def_format: ModuleDefFormat::Unknown,
+        is_external: false,
+        package_json: None,
```

Comment:
Side note: So using `this.load` API would cause the `package.json#sideEffects` being ignored. Because it skips the resolver the step, so we can't get a package json via oxc-resolver.

Reply:
Yeah. Here the rollup needs to manual set `moduleSideEffects` at here, the pr not including it.